splinepy/spline.py

```diff
@@ -1310,19 +1355,45 @@ def basis_and_support(self, queries, nthreads=None):
         basis: (n, prod(degrees + 1)) np.ndarray
         support: (n, prod(degrees + 1)) np.ndarray
         """
-        self._logd("Evaluating basis functions")
+        self._logd("Evaluating basis functions and support")
         queries = utils.data.enforce_contiguous(queries, dtype="float64")
 
         return super().basis_and_support(
             queries=queries,
             nthreads=_default_if_none(nthreads, settings.NTHREADS),
         )
 
+    @_new_core_if_modified
+    def basis_derivative(self, queries, orders, nthreads=None):
+        """
+        Returns derivative of basis functions of given queries.
+
```

Comment:
Can you add a small comment, that this is just evaluated on the support. Else, one could expect one basis function (derivative) for each ctps. With support in the function name that was clearer